From: Luca Weiss <luca.weiss@fairphone.com>
Date: Mon, 8 Apr 2024 11:27:53 +0200
Subject: Revert "dt-bindings: usb: add common Type-C USB Switch schema"

This seems to fix data-lanes validation, don't quite know why yet.
---
 .../devicetree/bindings/usb/fcs,fsa4480.yaml       | 13 +++--
 .../devicetree/bindings/usb/gpio-sbu-mux.yaml      | 18 +++---
 .../devicetree/bindings/usb/nxp,ptn36502.yaml      | 12 ++--
 .../devicetree/bindings/usb/onnn,nb7vpq904m.yaml   | 13 +++--
 .../bindings/usb/qcom,wcd939x-usbss.yaml           | 12 ++--
 .../devicetree/bindings/usb/usb-switch.yaml        | 67 ----------------------
 6 files changed, 37 insertions(+), 98 deletions(-)
 delete mode 100644 Documentation/devicetree/bindings/usb/usb-switch.yaml

diff --git a/Documentation/devicetree/bindings/usb/fcs,fsa4480.yaml b/Documentation/devicetree/bindings/usb/fcs,fsa4480.yaml
index 8b25b9a..3aa03fd 100644
--- a/Documentation/devicetree/bindings/usb/fcs,fsa4480.yaml
+++ b/Documentation/devicetree/bindings/usb/fcs,fsa4480.yaml
@@ -27,8 +27,13 @@ properties:
   vcc-supply:
     description: power supply (2.7V-5.5V)
 
-  mode-switch: true
-  orientation-switch: true
+  mode-switch:
+    description: Flag the port as possible handle of altmode switching
+    type: boolean
+
+  orientation-switch:
+    description: Flag the port as possible handler of orientation switching
+    type: boolean
 
   port:
     $ref: /schemas/graph.yaml#/$defs/port-base
@@ -74,9 +79,6 @@ required:
   - reg
   - port
 
-allOf:
-  - $ref: usb-switch.yaml#
-
 additionalProperties: false
 
 examples:
@@ -100,6 +102,7 @@ examples:
           port {
             fsa4480_ept: endpoint {
               remote-endpoint = <&typec_controller>;
+              data-lanes = <0 1>;
             };
           };
         };
diff --git a/Documentation/devicetree/bindings/usb/gpio-sbu-mux.yaml b/Documentation/devicetree/bindings/usb/gpio-sbu-mux.yaml
index 8a5f837..454dc7e 100644
--- a/Documentation/devicetree/bindings/usb/gpio-sbu-mux.yaml
+++ b/Documentation/devicetree/bindings/usb/gpio-sbu-mux.yaml
@@ -34,8 +34,13 @@ properties:
   vcc-supply:
     description: power supply
 
-  mode-switch: true
-  orientation-switch: true
+  mode-switch:
+    description: Flag the port as possible handle of altmode switching
+    type: boolean
+
+  orientation-switch:
+    description: Flag the port as possible handler of orientation switching
+    type: boolean
 
   port:
     $ref: /schemas/graph.yaml#/properties/port
@@ -49,15 +54,6 @@ required:
   - orientation-switch
   - port
 
-allOf:
-  - $ref: usb-switch.yaml#
-  - if:
-      required:
-        - mode-switch
-    then:
-      required:
-        - enable-gpios
-
 additionalProperties: false
 
 examples:
diff --git a/Documentation/devicetree/bindings/usb/nxp,ptn36502.yaml b/Documentation/devicetree/bindings/usb/nxp,ptn36502.yaml
index d805dde..eee548a 100644
--- a/Documentation/devicetree/bindings/usb/nxp,ptn36502.yaml
+++ b/Documentation/devicetree/bindings/usb/nxp,ptn36502.yaml
@@ -20,8 +20,13 @@ properties:
   vdd18-supply:
     description: Power supply for VDD18 pin
 
-  orientation-switch: true
-  retimer-switch: true
+  retimer-switch:
+    description: Flag the port as possible handle of SuperSpeed signals retiming
+    type: boolean
+
+  orientation-switch:
+    description: Flag the port as possible handler of orientation switching
+    type: boolean
 
   ports:
     $ref: /schemas/graph.yaml#/properties/ports
@@ -44,9 +49,6 @@ required:
   - compatible
   - reg
 
-allOf:
-  - $ref: usb-switch.yaml#
-
 additionalProperties: false
 
 examples:
diff --git a/Documentation/devicetree/bindings/usb/onnn,nb7vpq904m.yaml b/Documentation/devicetree/bindings/usb/onnn,nb7vpq904m.yaml
index 589914d..c0201da 100644
--- a/Documentation/devicetree/bindings/usb/onnn,nb7vpq904m.yaml
+++ b/Documentation/devicetree/bindings/usb/onnn,nb7vpq904m.yaml
@@ -21,8 +21,14 @@ properties:
     description: power supply (1.8V)
 
   enable-gpios: true
-  orientation-switch: true
-  retimer-switch: true
+
+  retimer-switch:
+    description: Flag the port as possible handle of SuperSpeed signals retiming
+    type: boolean
+
+  orientation-switch:
+    description: Flag the port as possible handler of orientation switching
+    type: boolean
 
   ports:
     $ref: /schemas/graph.yaml#/properties/ports
@@ -89,9 +95,6 @@ required:
   - compatible
   - reg
 
-allOf:
-  - $ref: usb-switch.yaml#
-
 additionalProperties: false
 
 examples:
diff --git a/Documentation/devicetree/bindings/usb/qcom,wcd939x-usbss.yaml b/Documentation/devicetree/bindings/usb/qcom,wcd939x-usbss.yaml
index 9634672..7ddfd331 100644
--- a/Documentation/devicetree/bindings/usb/qcom,wcd939x-usbss.yaml
+++ b/Documentation/devicetree/bindings/usb/qcom,wcd939x-usbss.yaml
@@ -35,8 +35,13 @@ properties:
   vdd-supply:
     description: USBSS VDD power supply
 
-  mode-switch: true
-  orientation-switch: true
+  mode-switch:
+    description: Flag the port as possible handle of altmode switching
+    type: boolean
+
+  orientation-switch:
+    description: Flag the port as possible handler of orientation switching
+    type: boolean
 
   ports:
     $ref: /schemas/graph.yaml#/properties/ports
@@ -58,9 +63,6 @@ required:
   - reg
   - ports
 
-allOf:
-  - $ref: usb-switch.yaml#
-
 additionalProperties: false
 
 examples:
diff --git a/Documentation/devicetree/bindings/usb/usb-switch.yaml b/Documentation/devicetree/bindings/usb/usb-switch.yaml
deleted file mode 100644
index da76118..00000000
--- a/Documentation/devicetree/bindings/usb/usb-switch.yaml
+++ /dev/null
@@ -1,67 +0,0 @@
-# SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
-%YAML 1.2
----
-$id: http://devicetree.org/schemas/usb/usb-switch.yaml#
-$schema: http://devicetree.org/meta-schemas/core.yaml#
-
-title: USB Orientation and Mode Switches Common Properties
-
-maintainers:
-  - Greg Kroah-Hartman <gregkh@linuxfoundation.org>
-
-description:
-  Common properties for devices handling USB mode and orientation switching.
-
-properties:
-  mode-switch:
-    description: Possible handler of altmode switching
-    type: boolean
-
-  orientation-switch:
-    description: Possible handler of orientation switching
-    type: boolean
-
-  retimer-switch:
-    description: Possible handler of SuperSpeed signals retiming
-    type: boolean
-
-  port:
-    $ref: /schemas/graph.yaml#/properties/port
-    description:
-      A port node to link the device to a TypeC controller for the purpose of
-      handling altmode muxing and orientation switching.
-
-  ports:
-    $ref: /schemas/graph.yaml#/properties/ports
-    properties:
-      port@0:
-        $ref: /schemas/graph.yaml#/properties/port
-        description:
-          Super Speed (SS) Output endpoint to the Type-C connector
-
-      port@1:
-        $ref: /schemas/graph.yaml#/$defs/port-base
-        description:
-          Super Speed (SS) Input endpoint from the Super-Speed PHY
-        unevaluatedProperties: false
-
-        properties:
-          endpoint:
-            $ref: /schemas/graph.yaml#/$defs/endpoint-base
-            unevaluatedProperties: false
-            properties:
-              data-lanes:
-                $ref: /schemas/types.yaml#/definitions/uint32-array
-                minItems: 1
-                maxItems: 8
-                uniqueItems: true
-                items:
-                  maximum: 8
-
-oneOf:
-  - required:
-      - port
-  - required:
-      - ports
-
-additionalProperties: true
